Tests/xSharePoint/xSharePoint.xSPSearchServiceApp.Tests.ps1

```diff
@@ -251,5 +251,83 @@ Describe "xSPSearchServiceApp" {
                 Test-TargetResource @testParams | Should Be $true
             }
         }
+        
+        $testParams.Add("SearchCenterUrl", "http://search.sp.contoso.com")
+        
+        Context "When the search center URL does not match" {
+            Mock Get-SPServiceApplication { 
+                return @(@{
+                    TypeName = "Search Service Application"
+                    DisplayName = $testParams.Name
+                    ApplicationPool = @{ Name = $testParams.ApplicationPool }
+                    Database = @{
+                        Name = $testParams.DatabaseName
+                        Server = @{ Name = $testParams.DatabaseServer }
+                    }
+                    SearchCenterUrl = "http://wrong.url.here"
+                })
+            }
+            Mock Get-SPServiceApplicationPool { return @{ Name = $testParams.ApplicationPool } }
+            Mock Get-SPEnterpriseSearchServiceInstance { return @{} }
+            Mock New-SPBusinessDataCatalogServiceApplication { }
+            Mock Start-SPEnterpriseSearchServiceInstance { }
+            Mock New-SPEnterpriseSearchServiceApplication { return @{} }
+            Mock New-SPEnterpriseSearchServiceApplicationProxy { }
+            Mock Set-SPEnterpriseSearchServiceApplication { } 
+            
+            Mock Get-SPWebApplication { return @(@{
+                Url = "http://centraladmin.contoso.com"
+                IsAdministrationWebApplication = $true
+            }) }
+            Mock Get-SPSite { @{} }
+            
+            Mock New-Object {
+                return @{
+                    DefaultGatheringAccount = "DOMAIN\username"
+                }
+            } -ParameterFilter { $TypeName -eq "Microsoft.Office.Server.Search.Administration.Content" }
+            
+            It "should return false from the test method" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+        }
```

Comment:
Thanks @ykuijs - Just pushed this one in now. I'll merge and close after the tests light up.